src/main/resources/com/fulcrumgenomics/umi/CollectDuplexSeqMetrics.R

```diff
@@ -0,0 +1,118 @@
+# CollectDuplexSeqMetrics tools.  All arguments are positional and required:
+#   1. The family size file with data on families by start/stop, ss and ds
+#   2. The duplex family size file
+#   3. The duplex yield metrics file
+#   4. The UMI metrics file
+#   5. An output PDF to write
+#   6+ One or more strings to use in plot titles to ID the sample/dataset
+#################################################################################
+
+options(warn = -1) # Don't emit warnings, only errors
+library(ggplot2)
+library(scales)  # Require by ggplot
+
+# Standard colors
+fgblue  = "#2E9DD7"
+fggreen = "#155936"
+fgred   = "firebrick3"
+fgcolors = c(fgblue, fggreen, fgred)
+
+args             = commandArgs(trailingOnly=T)
+familyData       = read.table(args[1], sep="\t", header=T)
+duplexFamilyData = read.table(args[2], sep="\t", header=T)
+yieldData        = read.table(args[3], sep="\t", header=T)
+umiData          = read.table(args[4], sep="\t", header=T)
+outputFile       = args[5]
+sampleInfo       = paste(args[6:length(args)])
+
+pdf(outputFile, width=11, height=8.5)
+
+# Plot #1 - Family Sizes for different kinds of familes
+ggplot(familyData) + aes(x=family_size) +
+  geom_line(aes(y=ds_count, color="DS Families")) +
+  geom_line(aes(y=ss_count, color="SS Families")) + 
+  geom_line(aes(y=cs_count, color="By Position")) + 
+  scale_x_continuous(trans="log2", minor_breaks=seq(0,max(familyData$family_size), by=2)) +
+  scale_color_manual(values=fgcolors) +
+  labs(x="Family Size (log2 scaled)", y="Count of Families", title=paste("Family Size Distributions for", sampleInfo)) +
+  theme(plot.title = element_text(hjust = 0.5), legend.title=element_blank())
+
+
+# Plot #2- Cumulative Family Sizes for different kinds of familes
+ggplot(familyData) + aes(x=family_size) +
+  geom_line(aes(y=cs_fraction_gt_or_eq_size, color="By Position"), alpha=0.5) + 
+  geom_line(aes(y=ss_fraction_gt_or_eq_size, color="SS Families"), alpha=0.5) + 
+  geom_line(aes(y=ds_fraction_gt_or_eq_size, color="DS Families"), alpha=0.5) +
+  scale_x_continuous(trans="log2", minor_breaks=seq(0,max(familyData$family_size), by=2)) +
+  scale_color_manual(values=fgcolors) +
+  labs(x="Family Size (log2 scaled)", y="Fraction of Families at >= Family Size", title=paste("Cumulative Family Size Distributions for", sampleInfo)) +
+  theme(plot.title = element_text(hjust = 0.5), legend.title=element_blank())
+
+# Plot #3a - Duplex Family Size Distribution
+ggplot(duplexFamilyData) +
+  aes(x=ab_size, y=ba_size, z=count) +
+  stat_summary_2d(fun=sum, binwidth=1) +
+  scale_fill_gradient2(low=fgblue, mid="white", high=fggreen, space="Lab",
+                       midpoint=log10(max(duplexFamilyData$count)), guide = "colourbar", trans="log2") +
+  labs(x="AB Reads", y="BA Reads", title=paste("Duplex Tag Family Size Distribution for", sampleInfo)) +
+  theme(plot.title = element_text(hjust = 0.5))
+
+# Plot #3b - Duplex Family Size Distribution for families with ab > 1 and ba > 1
```

Comment:
Heh.  I called them 3a and 3b since they're the same plot just with slightly different ranges of data displayed.